internal/test/wsu/wsu_test.go

```diff
@@ -139,20 +313,18 @@ func runWSUTestSuite(t *testing.T, vm e2ef.WindowsVM) {
 
 // testCNIConfig tests if the CNI config has required hostsubnet and servicenetwork CIDR
 // NOTE: split this into multiple tests when this grows
-func testCNIConfig(t *testing.T, vm e2ef.WindowsVM) {
+func testCNIConfig(t *testing.T, vm e2ef.WindowsVM, ansibleTempDir string) {
 	// Read the CNI config present on the Windows host
 	cniConfigFilePath := filepath.Join(ansibleTempDir, "cni", "config", "cni.conf")
 	cniConfigFileContents, err := readRemoteFile(cniConfigFilePath, vm)
 	require.NoError(t, err, "Could not get CNI config contents")
 
 	// Get the Windows node object
-	windowsNodes, err := framework.K8sclientset.CoreV1().Nodes().List(metav1.ListOptions{LabelSelector: windowsLabel})
-	require.NoError(t, err, "Could not get a list of Windows nodes")
-	require.Equalf(t, len(windowsNodes.Items), vmCount, "Expected %d Windows node(s) to be present but found %v",
-		vmCount, len(windowsNodes.Items))
+	node, err := getNode(vm.GetCredentials().GetIPAddress())
```

Comment:
Why are you making this call again here? You can pass the node object into this function.